config/initializers/omniauth.rb

```diff
@@ -1,64 +1,41 @@
-# Google's OAuth2 docs. Make sure you are familiar with all the options
-# before attempting to configure this gem.
-# https://developers.google.com/accounts/docs/OAuth2Login
 host = ENV['HOST'] || 'http://localhost:3000'
 
+# Omniauth will first try to use Google, and fall back to Okta if no 
+# GOOGLE_CLIENT_ID is defined.  If neither GOOGLE_CLIENT_ID nor
+# OKTA_CLIENT_ID is defined, the app will fail to boot
+#
 Rails.application.config.middleware.use OmniAuth::Builder do
-  # Default usage, this will give you offline access and a refresh token
-  # using default scopes 'email' and 'profile'
-  #
-  redirect_uri = "#{host}/auth/google_oauth2/callback"
-
-  options = {
-    scope:        'email,profile,calendar',
-    redirect_uri: redirect_uri,
-    setup: ->(env) do
-      env['omniauth.strategy'].options['token_params'] = {
-        redirect_uri: redirect_uri
-      }
+  raise 'Please configure either Google or SAML Authentication.' unless ENV['GOOGLE_CLIENT_ID'] || ENV['SAML_ISSUER']
+
+  if ENV['GOOGLE_CLIENT_ID']
+    # Default usage, this will give you offline access and a refresh token
+    # using default scopes 'email' and 'profile'
+
+    redirect_uri = "#{host}/auth/google_oauth2/callback"
+
+    options = {
+      scope:        'email,profile,calendar',
+      redirect_uri: redirect_uri,
+      setup: ->(env) do
+        env['omniauth.strategy'].options['token_params'] = {
+          redirect_uri: redirect_uri
+        }
+      end
+    }
+
+    if domain = ENV['OAUTH_DOMAIN']
+      options.merge!(hd: domain)
```

Comment:
single key merge is weird, could be 

```rb
options[:hd] = ENV['OAUTH_DOMAIN'] if ENV['OAUTH_DOMAIN']
```


or if you want to keep the if assignment, just set the key instead of merging